test/index.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const path = require('path');
+const { test } = require('ava');
+const webdriver = require('selenium-webdriver');
+
+const startGeckodriver = require('./util/start-geckodriver');
+
+let session, geckodriver;
+const firefoxArgs = process.env.CI ? [ '-headless' ] : [];
+
+test.before(async (t) => {
+  geckodriver = await startGeckodriver(1022, 12 * 1000);
+  session = new webdriver.Builder()
+    .usingServer('http://localhost:' + geckodriver.port)
+    .withCapabilities({
+      'moz:firefoxOptions': {
+        args: firefoxArgs
+      }
+    })
+    .forBrowser('firefox')
+    .build();
+  await session;
+});
+
+test.beforeEach((t) => {
+  t.context.session = session;
+  t.context.By = webdriver.By;
+  t.context.Key = webdriver.Key;
+  t.context.until = webdriver.until;
+});
+
+test.after.always(() => {
+  return Promise.resolve(session && session.close())
+    .then(() => geckodriver && geckodriver.stop());
+});
+
+/**
+ * Declare a test for a behavior documented on and demonstrated by an
+ * aria-practices examples page.
+ *
+ * @param {String} page - path to the example file
+ * @param {String} testId - unique identifier for the documented behavior
+ *                          within the demonstration page
+ * @param {Function} body - script which implements the test
+ */
+const ariaTest = (page, testId, body) => {
```

Comment:
As the primary interface for writing tests, I'd prefer it if this function did as little as necessary even if it means more code repetition inside the tests themselves. This will help keep parity with the Ava API, and make the tests more readable as discretely composed units.

Reply:
I hear where you are coming from -- I prefer readability to DRYness in tests myself. I wasn't particularly expecting to add much more functionality to ariaTest, but to be clear, do you think the subtest that currently exists in the ariaTest interface doesn't belong there, or are you just stating a preference not to see ariaTest grow too much?

Reply:
My concern here is because maintainers/users will primarily only ever be interacting with the unit tests, so they should ideally be able to understand everything that's happening from there without ever having to dig into ariaTest's internals.

One option would be to require ariaTest to be instantiated, similar to Webdriver's initialization process, and do the abstractions as part of that setup. Each page would require a declarative setup, but that can help with readability and DRYness.